src/studio/src/designer/backend.Tests/Controllers/TextsControllerTests.cs

```diff
@@ -49,6 +52,33 @@ public async Task GetText_ReturnsNorwegianBokmalText()
             Assert.Equal(expectedDictionary, responseDictionary);
         }
 
+        [Fact]
+        public async Task EditText_NewNbText()
+        {
+            var targetRepository = Guid.NewGuid().ToString();
+            await TestDataHelper.CopyRepositoryForTest("ttd", "new-resource-format", "testUser", targetRepository);
+
+            HttpClient client = GetTestClient();
+            string dataPathWithData = $"{_versionPrefix}/ttd/{targetRepository}/texts/nb";
+            HttpRequestMessage httpRequestMessage = new HttpRequestMessage(HttpMethod.Put, dataPathWithData);
+            httpRequestMessage.Content = JsonContent.Create(new { new_key_1 = "new_value_1", new_key_2 = "new_value_2" });
+            await AuthenticationUtil.AddAuthenticateAndAuthAndXsrFCookieToRequest(client, httpRequestMessage);
+
+            HttpResponseMessage response = await client.SendAsync(httpRequestMessage);
+            response.EnsureSuccessStatusCode();
+
+            try
+            {
+                string text = File.ReadAllText($"../../../_TestData/Repositories/testUser/ttd/{targetRepository}/App/config/texts/text.nb.json");
```

Comment:
Instead of reading this file from the test consider using either the GET endpoint of the controller or the TextsServices. That way you also get better test coverage on those and don't have to duplicate the code for reading.

Reply:
Smart. I am trying to fix this now, but I get errors with both controller method and service method... 
- With controller method I get errors from `AuthenticationHelper.GetDeveloperUserName(HttpContext)` when GetText tries to find the developer from the HttpContext.
- With service method I get error that ITextsSercice in the test constructor _not have matching fixture data_? 🤔